### PR TITLE
ARROW-12360: [Archery] Unify source code formatting of Archery 

### PR DESCRIPTION
**Pull request by kszucs**
Ran:
 - isort
 - black
 - flynt to prefer f-strings.

Included these check to the archery github action job and the pre-commit configuration.

### CODE REVIEW

**Comment by pitrou** [comment]
`archery lint --python` uses a different set of checks. It would be nicer IMHO if Archery followed the same rules as the rest of Python code.

**Comment by kszucs** [comment]
@pitrou I added a `archery lint --archery` command to run `isort`, `black` and `flake8`. I'd prefer black for archery since it has a stricter ruleset than autopep8 has.

**Comment by pitrou** [comment]
No problem with me.

**Comment by kszucs** [comment]
Perhaps ask others as well @jorisvandenbossche @xhochy

**Comment by jorisvandenbossche** [comment]
(I am in favor of using black more generally, so no objections to use it within the archery code ;))

**Comment by xhochy** [comment]
(I am in favor of using black more generally, so no objections to use it within the archery code ;))

File: dev/archery/archery/bot.py [hunk 1]
```diff
@@ -187,46 +189,60 @@ def _clone_arrow_and_crossbow(dest, crossbow_repo, pull_request):
...
+    "params",
+    multiple=True,
+    help="Additional task parameters for rendering the CI templates",
+)
+@click.option(
+    "--arrow-version",
+    "-v",
+    default=None,
+    help="Set target version explicitly.",
+)
```
Comment (pitrou) [inline dev/archery/archery/bot.py:236]: You can see here that `black` adds a lot of unwanted vertical size...

Comment (kszucs) [reply dev/archery/archery/bot.py:236]: I don't like that either, but I can tolerate that in exchange of automatic formatting.

Comment (pitrou) [reply dev/archery/archery/bot.py:236]: Right. The rest of our Python codebase uses autopep8, so you may want to try it too.

Comment (kszucs) [reply dev/archery/archery/bot.py:236]: Running autopep8 has changed 3 lines in double aggressive mode. 
The output of `black` seems more homogeneous to me and it is quicker as well. 

I'd still prefer the more stricter black over autopep8, but I don't insist to it.

**Comment by kszucs** [comment]
@pitrou any objections here?

**Comment by pitrou** [comment]
Well, as I said I'd rather see `autopep8` used like in the rest of the Python code.

**Comment by kszucs** [comment]
I'm not insisting on this then :)

**Pull request closed without merging**